codegen/chainconfig.go

```diff
@@ -1,143 +1,47 @@
 package codegen
 
-type ChainConfig struct {
-	ID               string // Public
-	DisplayName      string // Public
-	ExplorerLink     string
-	ApiEndpoint      string
-	FirehoseEndpoint string
-	Network          string
-	SupportsCalls    bool
+import (
+	"log"
+	"sync"
+
+	registry "github.com/pinax-network/graph-networks-libs/packages/golang/lib"
+)
+
+var (
+	registryNetworks     map[string]*registry.Network
+	registryNetworksOnce sync.Once
+)
+
+// GetRegistryNetworks returns a map of network ID to *registry.Network, using Pinax's registry as primary source.
+// Only networks with a Firehose endpoint are included.
+// If fetching fails, it falls back to loading from a local JSON file ("TheGraphNetworksRegistry.json").
+func GetRegistryNetworks() map[string]*registry.Network {
+	registryNetworksOnce.Do(func() {
+		reg, err := registry.FromLatestVersion()
+		if err != nil {
+			// Fallback: try to load from local file
+			// TODO: Validate where to put the registry file
+			reg, err = registry.FromFile("TheGraphNetworksRegistry.json")
+			if err != nil {
+				log.Fatalf("Failed to load registry from both network and file: %v", err)
+			}
+		}
+		m := make(map[string]*registry.Network)
+		for i, net := range reg.Networks {
+			if len(net.Services.Firehose) > 0 {
```

Comment:
Do I also need to filter out the ones where `len(net.Services.Subtreams) > 0`?

Reply:
The `FirehoseEndpoint` used in the code was misleading.